File: idb_companion/Utility/FBTestApplicationsPair.m
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "FBTestApplicationsPair.h"

@implementation FBTestApplicationsPair

#pragma mark Initializers

- (instancetype)initWithApplicationUnderTest:(FBInstalledApplication *)applicationUnderTest testHostApp:(FBInstalledApplication *)testHostApp
{
  self = [super init];
  if (!self) {
    return nil;
  }

  _applicationUnderTest = applicationUnderTest;
  _testHostApp = testHostApp;

  return self;
}

#pragma mark NSObject

- (NSString *)description
{
  return [NSString stringWithFormat:@"AUT %@, Test Host %@", self.applicationUnderTest, self.testHostApp];
}

@end

File: CompanionLib/Utility/FBTestApplicationsPair.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import FBControlCore
import Foundation

@objc public final class FBTestApplicationsPair: NSObject {

  @objc public let applicationUnderTest: FBInstalledApplication?
  @objc public let testHostApp: FBInstalledApplication?

  @objc public init(applicationUnderTest: FBInstalledApplication?, testHostApp: FBInstalledApplication?) {
    self.applicationUnderTest = applicationUnderTest
    self.testHostApp = testHostApp
    super.init()
  }

  public override var description: String {
    let autDesc = applicationUnderTest.map { "\($0)" } ?? "(null)"
    let hostDesc = testHostApp.map { "\($0)" } ?? "(null)"
    return "AUT \(autDesc), Test Host \(hostDesc)"
  }
}
